plugin/plugin.go

```diff
@@ -0,0 +1,107 @@
+package plugin
```

Comment:
The package should be called lib, not plugin. You may change the dir name, also.